package NSearchQuery;

enum EEmbedsSerializtionAlgo {
    // Continues = embeds lay one by one,
    // Discretization = each component is mapped into one byte
    // PantherModel =  used panther-terms modles specific seriazliation type
    //      for float <-> ui8 mapping
    PantherModelContinuesCommonRangeRenormDiscretization = 1;
     
}

message TEmbedsBundleProto {
    optional uint32 EmbedsCount = 1;
    optional uint32 EmbedsDimension = 2;
    optional bytes SeriazliedEmbeds = 3;
    optional EEmbedsSerializtionAlgo SerializationAlgo = 4;
    repeated float EmbedsWeights = 5 [packed = true];

    optional uint32 LbExpansionTypeTag = 10;
}

message TEmbedsBundleTransport {
    repeated TEmbedsBundleProto EmbedBundles = 1;
}
